addon/PreviewProvider.js

```diff
@@ -327,8 +342,12 @@ PreviewProvider.prototype = {
    */
   init() {
     this._alreadyRequested = new Set();
-    this._embedlyEndpoint = simplePrefs.prefs[EMBEDLY_PREF] + EMBEDLY_VERSION_QUERY + self.version;
+    this._metadataEndpoints = new Map();
     this.enabled = simplePrefs.prefs[ENABLED_PREF];
+    this._currentEndpoint = simplePrefs.prefs[CURRENT_ENDPOINT_PREF] + ENDPOINT_VERSION_QUERY + self.version;
```

Comment:
Can we move this logic into a method getCurrentEndpoint becuase we repeat it in the case statement above as well.